common/check_expired_locked_rules

```diff
@@ -6,51 +6,96 @@
 # You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 #
 # Authors:
-# - Cedric Serfon, <cedric.serfon@cern.ch>, 2015
+# - Cedric Serfon, <cedric.serfon@cern.ch>, 2020. 
+# - Eric Vaandering, <ewv@fnal.gov> 2020.
+# - Fernando Garzon, <oscar.fernando.garzon.miguez@cern.ch> 2020.
 
-'''
+"""
 Probe to check the locked expired rules or datasets with locked rules
-'''
+"""
+
+from __future__ import print_function
 
 import sys
-from rucio.db.sqla.session import get_session
+from datetime import datetime
+
+from rucio.core import monitor
+from rucio.db.sqla import models
+from rucio.db.sqla.session import BASE, get_session
+from sqlalchemy import func
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
+if BASE.metadata.schema:
+    schema = BASE.metadata.schema + '.'
+else:
+    schema = ''
+
 
 def main():
-    '''
+    """
     Probe to check the locked expired rules or datasets with locked rules
-    '''
+    """
     status = OK
     session = get_session()
     try:
-        query = "select rawtohex(id), scope, name, rse_expression from atlas_rucio.rules where locked=1 and expires_at<sys_extract_utc(localtimestamp)"
-        print 'Locked expired rules'
-        for row in session.execute(query):
+        print('Locked expired rules')
+        query = (session.query(models.ReplicationRule.id, models.ReplicationRule.scope,
+                               models.ReplicationRule.name, models.ReplicationRule.rse_expression)
+                 .filter(models.ReplicationRule.locked == 1)
+                 .filter(models.ReplicationRule.expires_at is not None)
+                 .filter(models.ReplicationRule.expires_at < datetime.utcnow()))
+        results = query.all()
+        for row in results:
             status = CRITICAL
-            print row[0], row[1], row[2]
+            rule_id = str(row[0])
+            print(rule_id, row[1], row[2])
+        print('Locked expired rule counts by RSE expression')
+        query = (session.query(models.ReplicationRule.scope, models.ReplicationRule.rse_expression,
+                               func.count(models.ReplicationRule.scope))
+                 .filter(models.ReplicationRule.locked == 1, models.ReplicationRule.expires_at <= datetime.utcnow())
+                 .group_by(models.ReplicationRule.scope, models.ReplicationRule.rse_expression))
+        for row in query.all():
+            print('judge.locked_expired_rules.%s.%s    %s' % (row[0], row[1], row[2]))
+            monitor.record_gauge(stat='judge.locked_expired_rules.%s.%s' % (row[0], row[1]), value=row[2])
```

Comment:
RSE expressions can be complex (e.g. `(istape=True&Tier=1)\site=IN2P3-CC`). Can’t this cause problems?

Reply:
We weren't even running this and when I do run it I don't find anything, so in the interest of getting this done, I deleted that section entirely. 

Fernando perhaps has an overlapping pull request just for this file and we can address it there.